R/expect_lint.R

```diff
@@ -49,7 +51,11 @@ expect_lint <- function(content, checks, ..., file = NULL, language = "en") {
     writeLines(content, con = file, sep = "\n")
   }
 
-  lints <- lint(file, ...)
+  if (!parse_settings) {
```

Comment:
That change is actually necessary to get the tests to pass with the new `.linter` config.
Previously, all tests used lintrs own `.lintr` file implicitly.
Now that we change it in this PR, a lot of tests break because some files are e.g. excluded.

The alternative would be to introduce a massive changeset to all of lintrs own tests (add `parse_settings = FALSE` everywhere and prepend `read_settings(NULL)` to all tests of all linters)
In favour of not touching every single test, I've opted to change `expect_lint` instead.

To see what happens without that, take a look at the CI build log before this fix:
https://travis-ci.org/github/jimhester/lintr/jobs/746322017#L1686

Reply:
WRT breaking user code: It will break code that relies on the `.lintr` file of a package being tested when using `expect_lint()`.
The only situation such a code might be used would be to verify that the linter settings are correct.

https://github.com/search?l=R&q=%22expect_lint%22&type=Code

Turns up a bunch of projects using code like this: https://github.com/2019-usfca-biol-422-fall/jbimperial-bioinformatics-project/blob/956dcc9b42ab0e9582dbc1763d90ffff8e5e733f/tests/render_rmds.R
If any of those had a `.lintr` (which they don't), that might cause unexpected behaviour.

I think it's safer to change all existing tests - although I haven't found any usage of `lintr::expect_lint` in combination with a custom `.lintr` file where it would break on github, but on the other hand packages using `expect_lint` to test their own custom linters would benefit from this default as well.
WDYT?

Reply:
Nevermind - we only have to update the tests in `test-knitr_formats.R` anyway. Done that and reverted the change.